mobx/example/todos/lib/todos.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:mobx/mobx.dart';
+
+class Todo {
+  Todo(String description) {
+    this.description = description;
+
+    markDone = action((bool flag) {
+      done = flag;
+    });
+
+    setDescription = action((String description) {
+      this.description = description;
+    });
+  }
+
+  // This boilerplate will be hidden away once the Code-generators are in place
+
+  //region Private Fields
+  final _description = Observable('');
+  final _done = Observable(false);
+  //endregion
+
+  String get description => _description.value;
+  set description(String value) => _description.value = value;
+
+  bool get done => _done.value;
+  set done(bool value) => _done.value = value;
+
+  Action setDescription;
+  Action markDone;
+}
+
+enum VisibilityFilter { all, pending, completed }
+
+class TodoList {
+  TodoList() {
+    _visibleTodos = Computed(() {
+      switch (filter) {
+        case VisibilityFilter.pending:
+          return pendingTodos;
```

Comment:
Could pending/completed todos just be calculated here? No need for extra Computed fields

Reply:
I kept it separate in order to leverage caching of the computed property. Else we would be recomputing every time we request `pendingTodos`